cohortextractor/mssql_utils.py

```diff
@@ -14,6 +14,15 @@
 
 logger = structlog.get_logger()
 
+SQLSERVER_TIMING_REGEX = re.compile(
+    r"""
+        .*SQL\sServer\s(?P<timing_type>parse\sand\scompile\stime|Execution\sTime).*  # SQl Server Statistics time message prefix
+        .*CPU\stime\s=\s(?P<cpu_time>\d+)\sms  # cpu time in ms
+        .*elapsed\stime\s=\s(?P<elapsed_time>\d+)\sms  # elapsed time in ms
+        """,
+    flags=re.S | re.X,
+)
+
```

Comment:
It is possible to get the statistics as XML. Not sure if thats better than regexing plain text though!

Reply:
Is it?  How do I do that?  Any idea if that would also get me the query it relates to?

Reply:
Also not sure if it's any better/easier!

Reply:
`SET STATISTICS XML ON`, possibly. Not sure its worth it. https://docs.microsoft.com/en-us/sql/t-sql/statements/set-statistics-xml-transact-sql?view=sql-server-ver16

It might also be a completely separate thing from `SET STATISTICS TIME ON`, so not sure if its worth pursuing

Reply:
I think it would be worth it - it looks like it might give us more details of the query - but I don't think there's a way to get it out of pymssql easily